matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/network/Request.kt

```diff
@@ -74,23 +74,26 @@ internal suspend inline fun <DATA> executeRequest(globalErrorReceiver: GlobalErr
 
             currentRetryCount++
 
-            if (exception is Failure.ServerError &&
-                    exception.httpCode == 429 &&
-                    exception.error.code == MatrixError.M_LIMIT_EXCEEDED &&
-                    currentRetryCount < maxRetriesCount) {
+            if (exception.isLimitExceededError() && currentRetryCount < maxRetriesCount) {
                 // 429, we can retry
-                delay(exception.getRetryDelay(1_000))
+                val retryDelay = exception.getRetryDelay(1_000)
+                if (retryDelay <= maxDelayBeforeRetry) {
```

Comment:
Maybe update the documentation of `maxDelayBeforeRetry` to inform that it also has an effect on 429 management.